v2/pkg/engine/resolve/context.go

```diff
@@ -14,6 +14,7 @@ import (
 
 type Context struct {
 	ctx              context.Context
+	Query            string
```

Comment:
Try to make it work without having to expose Query and set it from Router.

Reply:
Done